### ISSUE TITLE
1.5.4: handheld torch does not light properly

### ISSUE THREAD

**Issue by zkol** [anegostudios/VintageStory-Issues#127]
Observed in singleplayer, about 1,5 h after starting: Handheld torch does not illuminate the environment at some places (seems randomly, see screenshots)
![2018-05-27_23-36-33](https://user-images.githubusercontent.com/37601868/40591223-57847bb8-620d-11e8-94a2-b71726301e57.png)
![2018-05-27_23-36-49](https://user-images.githubusercontent.com/37601868/40591225-5d025f74-620d-11e8-9147-71130abf673e.png)
![2018-05-27_23-37-11](https://user-images.githubusercontent.com/37601868/40591227-61d70310-620d-11e8-91b4-b7c4aa4b46d9.png)
![2018-05-27_23-37-25](https://user-images.githubusercontent.com/37601868/40591232-68cc7664-620d-11e8-9d5a-602a11f68539.png)
Some minutes after his issue, the client starts lagging heavily (see next comment).

**Comment by zkol** [comment]
The lagging game seems to consume lots of memory, causing my computer to swap memory. After quitting, a mono process remains.
[client-main.txt](https://github.com/anegostudios/VintageStory-Issues/files/2043104/client-main.txt)
[client-debug.txt](https://github.com/anegostudios/VintageStory-Issues/files/2043105/client-debug.txt)
[server-main.txt](https://github.com/anegostudios/VintageStory-Issues/files/2043106/server-main.txt)
[server-debug.txt](https://github.com/anegostudios/VintageStory-Issues/files/2043107/server-debug.txt)

**Comment by copygirl** [comment]
Memory issue and process failing to close might both be separate issues. Then again, maybe the light issue is caused by lack of memory?

**Comment by zkol** [comment]
Hard to tell. But I try to describe what I observed and what I guessed after looking into the logfiles to find a correlation of logged events with actions an observations I made.

I noticed the bug at least two or three times before, always without the extensive lagging and fail to close problem (alone in singleplayer and also when sharing singleplayer via LAN). But in this cases I avoided to stress this bug by using the torch in the dark and used the next convenient situation to leave and join again to get rid of the bug (which was about 15 mins in the case of the LAN session).

But for the bug report it was different, I tried to observe the bug, doing the repetitive standard stuff (flint-knapping axes, felling bamboos, harvesting, planting, frequently checking if the bug appears). And as the bug appeared, I continued using the "disfunct" torch while typing to discord, and while wandering around in the dark, observing that torch lighting "switched" to work again without leaving/joining. But this was randomly only on certain blocks.  So I tried again and again, taking screenshots.

I had screenshots before so I won't blame the screenshots. But I got the feeling that the things went really bad with lagging after stressing the "switching" point, when lighting was enabled, disabled, enabled, etc. in a high frequency. 

I noticed the bug about 23:30 the first time and there was a short server overload a few minutes before in the log (23:25). Maybe there is some kind of correlation. And there is a lot of server overload in the logs after documentation of the "switching" by screenshots (after 23:37). 

This was the situation were I was curious and tried out what might trigger the light "switch" of my torch. It was randomly: on ... off .. on ... off depending on the block were I stood. There was maybe a kinda correlation with passing hit boxes of pumpkins, ladders, or something else that is invsible. But thats really guessing, it seemed random to me, I  really got no pattern.

But it seemed that the lagging started and got worse the more often I "switched" after a certain point. So I got the impression that there might be a correlation between the memory issue and the lighting bug.
